### DISCUSSION TITLE
Dashboard Weekly Updates [05/02/24 - 12/02/24]

### DISCUSSION THREAD

**Discussion by joshenlim** [supabase/supabase#21219] [Changelog]
## SQL editor bulk deletes

https://github.com/supabase/supabase/assets/19742402/f50eabbf-9c30-4828-8f5f-0efeef67a025

We hear you! This has been a very popular request by everyone and we're happy to make the first step to improving the UX around your SQL snippets. You can now delete your queries in bulk - gone are the days of rows full with Untitled queries 😄 Fret not, we're also aware that everyone is also requesting for better organization of snippets (specifically folders) - we're actively figuring out how best to bring that UX into the dashboard for everyone so be sure to watch this space 😉

PR: [https://github.com/supabase/supabase/pull/20927](https://github.com/supabase/supabase/pull/20927)

Link: [https://supabase.com/dashboard/project/_/sql/new](https://supabase.com/dashboard/project/_/sql/new)

## Query performance updates

<img src="https://github.com/supabase/supabase/assets/37541088/4d13c4bb-a747-4790-b969-9fb162fabc9a" width="500"/>

Unoptimized queries are a major cause of poor database performance - which is why the query performance report was initially built. We're equipping this page with better tooling, allowing users to:

1. Search by query or role
2. Sort results by latency
3. Expand results to view the full query that was run

As always, if there's anything more we can do for you, feel free to give us a shout in the feedback widget up top 🙂

PR: [https://github.com/supabase/supabase/pull/20907](https://github.com/supabase/supabase/pull/20907)

Link: [https://supabase.com/dashboard/project/_/reports/query-performance](https://supabase.com/dashboard/project/_/reports/query-performance)

## Logs Explorer face lift

<img src="https://github.com/supabase/supabase/assets/37541088/2b92056f-1278-4132-8688-2bd52898177b" width="500" />

In an effort to make our UI more consistent + coherent across all products, we've revamped the Logs Explorer to look just like the SQL Editor in hopes that there's less UI for users to learn, and users can just stay focused on doing what they want to do.

PR: [https://github.com/supabase/supabase/pull/21055](https://github.com/supabase/supabase/pull/21055)

Link: [https://supabase.com/dashboard/project/_/logs/explorer](https://supabase.com/dashboard/project/_/logs/explorer)

## Support for composite foreign keys in table editor

<img src="https://github.com/supabase/supabase/assets/19742402/12f8efb0-8190-4680-82d7-0f19c3c8d7b2" width="500" />

The previous UI for managing foreign keys in the table editor had functional limitations as it assumed single column relations for foreign keys (overly simplified). We've thus shifted the management of foreign keys into the table side panel editor instead. You can manage all foreign keys across all columns on the table in one place, rather than going into each column individually to do so.

PR: [https://github.com/supabase/supabase/pull/21078](https://github.com/supabase/supabase/pull/21078)

Link: [https://supabase.com/dashboard/project/_/editor](https://supabase.com/dashboard/project/_/editor)

### Table Editor
- Fixed duplicating a table not saving it's description ([PR](https://github.com/supabase/supabase/pull/21215))
- Fixed viewing a reference row in the table editor from the grid not updating the selected schema if referenced row is in another schema ([PR](https://github.com/supabase/supabase/pull/21212))
- Fixed errors from adding data via spreadsheet/CSV text not surfacing as toasts ([PR](https://github.com/supabase/supabase/pull/21137))

### SQL Editor
- Fixed long errors getting cut off / not horizontally scrollable ([PR](https://github.com/supabase/supabase/pull/21213))
- Fixed deleting a query that's not saved throws an error ([PR](https://github.com/supabase/supabase/pull/21082))

### Logs Explorer
- Inserting a source will insert the value after the `FROM` command ([PR](https://github.com/supabase/supabase/pull/21114))

### Database
- Fixed enumerated types side panel input fields not resetting after saving ([PR](https://github.com/supabase/supabase/pull/21107))
- Fixed roles side panel input fields not resetting after saving ([PR](https://github.com/supabase/supabase/pull/20840))

### Misc
- Support page is now mobile responsive ([PR](https://github.com/supabase/supabase/pull/20999))

**Comment by csjasonk** [comment]
All of my saved queries are GONE gah! 

I tested adding a new query, and it makes the "Your Queries" appear... but my existing queries are not there... I'm scared to create/save a new one for fear it will actually erase the old ones.. which seem to be continuing to run even though they are not visible/editable.
